### PR TITLE
Reach Full Unit Testing Coverage for HLP 

### PR DESCRIPTION
**Pull request by LucioDonda**
|Related issue|
|---|
|#12182|

## Description

* Clean Unit Testing for HLP. 
* Unifying test names.
* Creating additional tests.

### CODE REVIEW

File: src/engine/test/source/hlp/hlp_test.cpp [hunk 1]
```diff
@@ -120,44 +122,135 @@ TEST(hlpTests, url_parsing)
     ASSERT_EQ("user", result["url.username"]);
 }
 
-TEST(json_test, dummy_test)
+// Test: parsing IP addresses
+TEST(hlpTests_IPaddress, IPaddress_success)
```
Comment (palaciosjeremias) [inline src/engine/test/source/hlp/hlp_test.cpp:126]: We should refactor and rename this test and the below one to only test IPV4 formats. As IPV6 has its own tests.

File: src/engine/test/source/hlp/hlp_test.cpp [hunk 2]
```diff
@@ -120,44 +122,135 @@ TEST(hlpTests, url_parsing)
...
     auto parseOp = getParserOp(logQl);
     auto result = parseOp(event);
 
+    ASSERT_EQ(true, static_cast<bool>(parseOp));
     ASSERT_EQ("{\"String\":\"This is a string\"}", result["_json1"]);
     ASSERT_EQ("{\"String\":\"This is another string\"}", result["_json2"]);
 }
 
-TEST(map_test, success_test)
+TEST(hlpTests_json, failed)
```
Comment (palaciosjeremias) [inline src/engine/test/source/hlp/hlp_test.cpp:190]: We should rename this test to something like "failed_incomplete_json"

File: src/engine/test/source/hlp/hlp_test.cpp [hunk 3]
```diff
@@ -120,44 +122,135 @@ TEST(hlpTests, url_parsing)
...
+    auto parseOp = getParserOp(logQl);
+    auto result = parseOp(event);
+
+    ASSERT_EQ(true, static_cast<bool>(parseOp));
+    ASSERT_EQ("{\"String\":[{\"SecondString\":\"This is a string\"},{\"ThirdString\":\"This is a string\"}]}", result["_json1"]);
+}
+
+TEST(hlpTests_json, failed_not_string)
+{
+    const char *logQl = "<_json2/JSON>";
```
Comment (palaciosjeremias) [inline src/engine/test/source/hlp/hlp_test.cpp:216]: t will be more clear if we avoid numbering the temporal field as it isn´t necessary (even in the above tests)

File: src/engine/test/source/hlp/hlp_test.cpp [hunk 4]
```diff
@@ -417,12 +552,26 @@ TEST(hlpTimestampTests, specific_format)
     ASSERT_EQ("5", result["_stampTs.seconds"]);
 }
 
-TEST(domain_test, success)
+TEST(hlpTests_Timestamp, kitchen)
```
Comment (palaciosjeremias) [inline src/engine/test/source/hlp/hlp_test.cpp:555]: As this test is known to fail, we should add a comment explaining this

**Review by palaciosjeremias — CHANGES_REQUESTED** [review]

File: src/engine/test/source/hlp/hlp_test.cpp (hunk 4 above)
Comment (LucioDonda) [reply src/engine/test/source/hlp/hlp_test.cpp:544]: done

File: src/engine/test/source/hlp/hlp_test.cpp (hunk 3 above)
Comment (LucioDonda) [reply src/engine/test/source/hlp/hlp_test.cpp]: done

File: src/engine/test/source/hlp/hlp_test.cpp (hunk 2 above)
Comment (LucioDonda) [reply src/engine/test/source/hlp/hlp_test.cpp]: done

File: src/engine/test/source/hlp/hlp_test.cpp (hunk 1 above)
Comment (LucioDonda) [reply src/engine/test/source/hlp/hlp_test.cpp]: done

File: src/engine/test/source/hlp/hlp_test.cpp [hunk 5]
```diff
@@ -71,33 +72,33 @@ TEST(hlpTests, invalid_logql)
...
     auto result = parseOp(event);
 
+    ASSERT_EQ(true, static_cast<bool>(parseOp));
     ASSERT_EQ("temp", result["_temp"]);
     ASSERT_EQ("temp1", result["_temp1"]);
     ASSERT_EQ("temp2", result["_temp2"]);
 }
 
-TEST(hlpTests, url_parsing)
+TEST(hlpTests_logQL, url_parsing)
```
[the remark this answers is not in the document]
Comment (LucioDonda) [reply src/engine/test/source/hlp/hlp_test.cpp]: done

File: src/engine/test/source/hlp/hlp_test.cpp [hunk 6]
```diff
@@ -2,61 +2,62 @@
 
 #include <hlp/hlp.hpp>
 
-static const char *logQl =
-    "<source.address> - <json> - [<timestamp/APACHE>] \"<http.request.method> <url> HTTP/<http.version>\" "
+
+// Test: Parsing logQL expressions
```
[the remark this answers is not in the document]
Comment (LucioDonda) [reply src/engine/test/source/hlp/hlp_test.cpp:6]: done

File: src/engine/test/source/hlp/hlp_test.cpp [hunk 7]
```diff
@@ -2,61 +2,62 @@
...
+    ASSERT_EQ("query=%22a%20query%20with%20a%20space%22", result["url.query"]);
+    ASSERT_EQ("https", result["url.scheme"]);
+    ASSERT_EQ("user", result["url.username"]);
 
-    fprintf(stderr, "\n--------------\n\n");
 }
 
-// Test: parsing succesfully three different IP addresses
-TEST(hlpTests, IP_Parser)
+TEST(hlpTests_logQL, logQL_general2)
```
[the remark this answers is not in the document]
Comment (LucioDonda) [reply src/engine/test/source/hlp/hlp_test.cpp]: done

File: src/engine/test/source/hlp/hlp_test.cpp [hunk 8]
```diff
@@ -2,61 +2,62 @@
 
 #include <hlp/hlp.hpp>
 
-static const char *logQl =
-    "<source.address> - <json> - [<timestamp/APACHE>] \"<http.request.method> <url> HTTP/<http.version>\" "
+
+// Test: Parsing logQL expressions
+TEST(hlpTests_logQL, logQL_general)
+{
+    const char *logQl =
```
[the remark this answers is not in the document]
Comment (LucioDonda) [reply src/engine/test/source/hlp/hlp_test.cpp]: done

File: src/engine/test/source/hlp/hlp_test.cpp [hunk 9]
```diff
@@ -2,61 +2,62 @@
...
+    ASSERT_EQ("1", result["timestamp.month"]);
+    ASSERT_EQ("2", result["timestamp.day"]);
+    ASSERT_EQ("15", result["timestamp.hour"]);
+    ASSERT_EQ("4", result["timestamp.minutes"]);
+    ASSERT_EQ("0", result["timestamp.seconds"]);
+    ASSERT_EQ("-0700", result["timestamp.timezone"]);
 }
 
-TEST(hlpTests, invalid_logql)
+TEST(hlpTests_logQL, invalid_logql)
```
[the remark this answers is not in the document]
Comment (LucioDonda) [reply src/engine/test/source/hlp/hlp_test.cpp]: renamed

File: src/engine/test/source/hlp/hlp_test.cpp [hunk 10]
```diff
@@ -2,61 +2,62 @@
...
-TEST(hlpTests, IP_Parser)
+TEST(hlpTests_logQL, logQL_general2)
 {
-    const char *logQl = "<source.ip> - <server.ip> -- <source.nat.ip> \"-\" \"-\"";
-    const char *event = "127.0.0.1 - 2001:db8:3333:4444:CCCC:DDDD:EEEE:FFFF -- 0:db8:0:4:CCCC:0:EEEE:FFFF \"-\" \"-\"";
-
-    fprintf(stderr, "\n\n---HLP IP parser Test---\n");
+    const char *logQl = "<source.ip> - - [<timestamp/RFC822Z>] \"-\" "
+                            "<http.response.status_code> <http.response.body.bytes> \"-\" \"-\"";
+    const char *event = "127.0.0.1 - - [02 Jan 06 15:04 -0700] \"-\" 408 152 \"-\" \"-\"";
```
[the remark this answers is not in the document]
Comment (LucioDonda) [reply src/engine/test/source/hlp/hlp_test.cpp]: no, indeed, I removed it

File: src/engine/test/source/hlp/hlp_test.cpp [hunk 11]
```diff
@@ -71,33 +72,33 @@ TEST(hlpTests, invalid_logql)
     ASSERT_EQ(false, static_cast<bool>(invalidFunc3));
 }
 
-TEST(hlpTests, optional_Or)
+TEST(hlpTests_logQL, optional_Or)
```
[the remark this answers is not in the document]
Comment (LucioDonda) [reply src/engine/test/source/hlp/hlp_test.cpp:62]: done

File: src/engine/test/source/hlp/hlp_test.cpp [hunk 12]
```diff
@@ -71,33 +72,33 @@ TEST(hlpTests, invalid_logql)
...
     static const char *event = "this won't match an url 127.0.0.1 but will match ip";
 
     auto parseOp = getParserOp(logQl);
     auto result = parseOp(event);
 
-    fprintf(stderr, "Matched: [%s]\n", result["source.ip"].c_str());
-
+    ASSERT_EQ(true, static_cast<bool>(parseOp));
     ASSERT_EQ("127.0.0.1", result["source.ip"]);
 }
```
[the remark this answers is not in the document]
Comment (LucioDonda) [reply src/engine/test/source/hlp/hlp_test.cpp:73]: done

File: src/engine/test/source/hlp/hlp_test.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (LucioDonda) [reply src/engine/test/source/hlp/hlp_test.cpp]: done

**Review by palaciosjeremias — APPROVED** [review]
LGTM

**Pull request merged**